### DISCUSSION TITLE
I made a 3D visualisation tool for telemetry data from quali at Albert Park in Melbourne

### DISCUSSION THREAD

**Discussion by david-arcus** [theOehrly/Fast-F1#362] [Show and tell]
https://user-images.githubusercontent.com/6078748/235623550-7b3a1504-ce07-44ea-9c99-93e746e7e84b.mp4

Hi everyone – I thought I'd share a tool I've been working on that simply wouldn't be possible without FastF1. This is a Unity app that takes the telemetry and position data provided by FastF1, converts it into splines, then animates the position of the cars. The 3D data for Albert Park was gathered from OpenStreetMaps and the track itself is extruded from an SVG from Wikipedia. Happy to answer any questions.

**Comment by theOehrly** [comment]
Nice work, that looks amazing 👍

**Comment by MooshuBeef** [comment]
AMAZING! When you say position data which API calls did you use specifically?

**Comment by david-arcus** [comment]
Thanks! Nothing too elaborate here, FastF1 makes it very easy to get what you need. I just get the telemetry and lap data for each driver's fastest lap and then dump it to JSON, which is then read by the Unity app:

```
session = fastf1.get_session(2023, "Miami", "Qualifying")
laps = session.load_laps(with_telemetry=True)

drivers = {}

for driver in session.drivers:
    drivers[driver] = []

    lap = laps.pick_driver(driver).pick_fastest()
    tel = lap.get_telemetry()

    drivers[driver].append(tel.to_json()) # telemetry data (includes interpolated position data, speed etc.)
    drivers[driver].append(lap.to_json()) # lap data (e.g sector times)
    drivers[driver].append(session.get_driver(driver).to_json()) # driver information (e.g team colour etc.)
```

This is a slightly simplified version as you'll want to do things like cache the data and check if a lap time exists for each driver.
